packages/frontend/src/features/shared/hooks/useDatePicker.ts

```diff
@@ -0,0 +1,77 @@
+import { useUserState } from '@/features/core'
+import { useCallback, useState } from 'react'
+import {
+    EpochDateService,
+    FromToEpoch,
+    InvalidFromToEpoch,
+} from '../services/EpochDateService'
+
+export function useDatePicker() {
+    const { userState } = useUserState()
+    const [startDate, setStartDate] = useState<undefined | Date>(undefined)
+    const [endDate, setEndDate] = useState<undefined | Date>(undefined)
+    const [isDateSelected, setIsDateSelected] = useState(false)
+    const [fromToEpoch, setFromToEpoch] = useState<FromToEpoch>(
+        new InvalidFromToEpoch(),
+    )
+    const updateFromToEpoch = useCallback(async () => {
+        if (!userState) {
+            setFromToEpoch(new InvalidFromToEpoch())
+            return
+        }
+        setFromToEpoch(
+            EpochDateService.createFromToEpochByDateRange(
+                startDate,
+                endDate,
+                userState.sync,
+            ),
+        )
+    }, [startDate, endDate, userState])
+    const onChange = (dates: any) => {
+        const [start, end] = dates
+        setStartDate(start)
+        setEndDate(end)
+        setIsDateSelected(!!start && !!end)
+    }
+    const setToday = () => {
+        const today = new Date()
+        setStartDate(today)
+        setEndDate(today)
+        setIsDateSelected(true)
+    }
```

Comment:
There is an issue in this implementation. `fromEpoch` and `toEpoch` would be the same epoch number if `startDate` is equal to `endDate`. However, `fromEpcoh` should be the start epoch of today, and `toEpoch` should be the end epoch of today. `onChange`, `setPast7Days` and `setPast30Days` have the same issue.

`startDate` should be set to the start of today, and `endDate` should be the end of today. You can use day.js to implement this. For example,

```js
const now = dayjs()
setStartDate(now.startOf('day').toDate())
setEndDate(now.endOf('day').toDate())
```

Or you can fix this issue in `EpochDateService`.